src/imfilter.jl

```diff
@@ -6,17 +6,25 @@ function imfilter_cpu(img, krn)
   imfilter(img, centered(krn), Inner(), Algorithm.FFT())
 end
 
+img_to_fftimg_gpu = IdDict()
```

Comment:
Hi @ClaroHenrique can you please comment on the dictionary approach? I was wondering if we could refactor the code without global dictionaries. Basically as we discussed over Zulip. We may need to refactor the preprocessing function to simply move the training image to the CuArray once.